js/CRM_Contact_Page_View_UserDashBoard.js

```diff
@@ -17,5 +17,33 @@
     var html = $(el).html().replace(/[\[\]]/g , '');
     $(el).html(html);
   })
+
+  // Change renew button link to match the users current membership type
+  $(document).ready(function(){
+    $('.form-item .secondary-navigation').each(function(){
+      // Get the membership type
+      var memberShipType = $(this).parent('td').parent('tr').find('td:first-child').html();
+      var newHref;
+
+      if(
+        memberShipType === 'New Discerning Member - 1 year (CANADA)'
+        || memberShipType === 'New Discerning Couple Member - 1 year (CANADA)'
+        || memberShipType === 'Individual Member - 1 year (Canada Renewal)'
+        || memberShipType === 'Couple Member - 1 year (Canada Renewal)'
+      ) {
+        newHref = $('a', this).attr('href').replace('id=10', 'id=14');
```

Comment:
Just noticed this problem: The URL can contain `id=10` in places that might not be expected, for example the `mid` query parameter, and we can't guarantee the order in which they appear, so this might make trouble. Example: `http://example.com/login/?page=CiviCRM&q=civicrm%2Fcontribute%2Ftransact&id=14&mid=10123&reset=1`. If the `mid` parameter is `mid=10123` (and if the `id` parameter doesn't come first), this code would change `mid` to `mid=14123` and would not change the `id` parameter at all. We need to ensure it only changes the `id` parameter -- maybe use a regex?

Reply:
Done